### PR TITLE
feat: introduce new translator role

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1779
https://oat-sa.atlassian.net/browse/ADF-1780
https://oat-sa.atlassian.net/browse/ADF-1781
https://oat-sa.atlassian.net/browse/ADF-1783
https://oat-sa.atlassian.net/browse/HKD-6

# Goal 

Introduce new translator role that will be able to perform translations actions

# PRs
* https://github.com/oat-sa/tao-core/pull/4094
* https://github.com/oat-sa/extension-tao-item/pull/660
* https://github.com/oat-sa/extension-tao-itemqti/pull/2571
* https://github.com/oat-sa/extension-tao-testqti/pull/2503

# TODO 

- [ ] Update composer dependencies

### CODE REVIEW

File: models/ontology/taotest.rdf [hunk 1]
```diff
@@ -92,6 +92,22 @@
     <rdfs:comment xml:lang="en-US"><![CDATA[The Test Translator Role]]></rdfs:comment>
   </rdf:Description>
 
+  <rdf:Description rdf:about="http://www.tao.lu/Ontologies/TAO.rdf#TranslationCompletionStatuses">
```
Comment (gabrielfs7) [inline models/ontology/taotest.rdf:95]: @shpran I think for this one we need to add in the test namespace `http://www.tao.lu/Ontologies/TAOTest.rdf`

File: models/ontology/taotest.rdf [hunk 2]
```diff
@@ -107,4 +123,13 @@
   <rdf:Description rdf:about="http://www.tao.lu/Ontologies/TAO.rdf#TranslationProgress">
     <rdfs:domain rdf:resource="http://www.tao.lu/Ontologies/TAOTest.rdf#Test"/>
   </rdf:Description>
+  <rdf:Description rdf:about="http://www.tao.lu/Ontologies/TAO.rdf#TranslationCompletion">
```
Comment (gabrielfs7) [inline models/ontology/taotest.rdf:126]: @shpran I think the same as here https://github.com/oat-sa/extension-tao-test/pull/476/files/556812e3dc2a3e479a6e837f9aff29e28846a30b..2c62e4d30448e64b2f8f702ca4d8a631870d8e4b#r1750050841

**Review by shpran — APPROVED** [review]
Good job! 👍🏻

File: models/classes/Translation/Listener/TestCreatedEventListener.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+use oat\generis\model\data\Ontology;
+use oat\oatbox\event\Event;
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\tao\model\TaoOntology;
+use oat\taoItems\model\event\ItemCreatedEvent;
+use oat\taoTests\models\event\TestCreatedEvent;
+use Psr\Log\LoggerInterface;
+
+class TestCreatedEventListener
```
Comment (gabrielfs7) [inline models/classes/Translation/Listener/TestCreatedEventListener.php:33]: - [ ] @shpran Can you please add missing unit test for this one? Thank you!

File: models/classes/Translation/Listener/TestCreatedEventListener.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        FeatureFlagCheckerInterface $featureFlagChecker,
+        Ontology $ontology,
+        LoggerInterface $logger
+    ) {
+        $this->featureFlagChecker = $featureFlagChecker;
+        $this->ontology = $ontology;
+        $this->logger = $logger;
+    }
+
+    public function populateTranslationProperties(Event $event): void
```
Comment (gabrielfs7) [inline models/classes/Translation/Listener/TestCreatedEventListener.php:49]: - Due the class name `TestCreatedEventListener`, seems we will not accept any other type of event. I would make it static in this case.

cc: @shpran 

```suggestion
    public function populateTranslationProperties(TestCreatedEvent $event): void
```

File: models/classes/Translation/Listener/TestCreatedEventListener.php (hunk 3 above)
Comment (gabrielfs7) [reply models/classes/Translation/Listener/TestCreatedEventListener.php:31]: Closing as done

File: models/classes/Translation/Listener/TestCreatedEventListener.php (hunk 4 above)
Comment (gabrielfs7) [reply models/classes/Translation/Listener/TestCreatedEventListener.php]: Closing as done

File: models/ontology/taotest.rdf (hunk 1 above)
Comment (shpran) [reply models/ontology/taotest.rdf]: Done ✅

**Pull request merged**